models/basgra/R/run_BASGRA.R

```diff
@@ -0,0 +1,268 @@
+
+##-------------------------------------------------------------------------------------------------#
+##' BASGRA wrapper function. Runs and writes model outputs in PEcAn standard.
+##'
+##' BASGRA is written in fortran is run through R by wrapper functions written by Marcel Van Oijen.
+##' This function makes use of those wrappers but gives control of datastream in and out of the model to PEcAn.
+##' With this function we skip model2netcdf, we can also skip met2model but keeping it for now. 
+##' write.config.BASGRA modifies args of this function through template.job
+##' then job.sh runs calls this function to run the model
+##'
+##' @name run_BASGRA
+##' @title run BASGRA model
+##' @param run_met path to climate file, should change when I get rid of met2model?
+##' @param run_params parameter vector
+##' @param start_date start time of the simulation
+##' @param end_date end time of the simulation
+##' @param outdir where to write BASGRA output
+##' @param sitelat latitude of the site
+##' @param sitelon longitude of the site
+##' 
+##' @export
+##' @useDynLib PEcAn.BASGRA
+##' @author Istem Fer
+##-------------------------------------------------------------------------------------------------#
+
+run_BASGRA <- function(run_met, run_params, start_date, end_date, outdir, sitelat, sitelon){
+
+  start_date  <- as.POSIXlt(start_date, tz = "UTC")
+  end_date    <- as.POSIXlt(end_date, tz = "UTC")
+  start_year  <- lubridate::year(start_date)
+  end_year    <- lubridate::year(end_date)
+  
+
+
+  ################################################################################
+  ### FUNCTIONS FOR READING WEATHER DATA
+  mini_met2model_BASGRA <- function(file_path,
+                                    start_date, start_year,
+                                    end_date, end_year) {
+
+    out.list <- list()
+    
+    ctr <- 1
+    for(year in start_year:end_year) {
+      
+      diy <- PEcAn.utils::days_in_year(year)
+      
+      NWEATHER              <- as.integer(8)
+      matrix_weather        <- matrix( 0., nrow=diy, ncol=NWEATHER )
+      
+      
+      # prepare data frame for BASGRA format, daily inputs, but doesn't have to be full year
+      
+      
+      matrix_weather[ ,1] <- rep(year, diy) # year
+      matrix_weather[ ,2] <- seq_len(diy) # day of year, simple implementation for now
+      
+      old.file <- file.path(dirname(file_path), paste(basename(file_path), year, "nc", sep = "."))
+      
+      if (file.exists(old.file)) {
+        
+        ## open netcdf
+        nc <- ncdf4::nc_open(old.file)  
+        
```

Comment:
This will save you having to close the file at the end of the function, and has the nice side effect of calling this if the function exits early for any reason, including throwing an error.

```suggestion
on.exit(ncdf4::nc_close(nc), add = TRUE)
```